src/input.hh

```diff
@@ -20,12 +21,14 @@ struct flags {
         disassemble(false),
         hide_nops(false),
         expand(false),
-        null_terminated(false) {}
+        null_terminated(false),
+        compile(false) {}
 
     constexpr bool is_valid() const noexcept {
         return !(
             (show_stack && !debug) || ((debug || warnings || pipekill) && disassemble) || (hide_nops && !disassemble)
             || (expand && (debug || warnings || pipekill || disassemble))
+            || (compile && (debug || warnings || pipekill || disassemble || expand))
```

Comment:
I feel like some of these could be 2- or 3-bit enums rather than individual booleans, but it's not a space concern yet. See the `static_assert` in thread.hh

Reply:
Thinking about this more, I'm leaning towards "make illegal states unrepresentable".

Reply:
It's not that simple: `-s` requires `-d`, but one could say `-s -d` rather than `-ds`.